iocore/nexthop/NextHopHealth.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include "NextHopHost.h"
+
+/// ----------------------------------
+/// Health Check plugin API
+/// TODO: move to another file
+
+/// Define the types of health checks, or reasons to be down.
+enum HealthCheck_t : uint8_t {
```

Comment:
amc: enum class HealthCheck and drop the HC_ prefix? In general, TS doesn't use _t for classes (in fact, I think it is forbidden).

Reply:
I was a was avoiding using enum class, because converting them to ints on every use case is annoying. 
I'll revisit this now that I have the PropertyBlocks better defined.